Adaptive Trapezoid C++ implementation

abruns123 opened the pull request:
## Linting
 The python code should lint

## Contents
Added c++ implementation of the adaptive trapezoid integration to the c++ calculus file. 
Also added pytest for that and adjusted wrapper implementation of adaptive trapezoid to take 
in general functions. 

## Labels and Milestones


## Reviewer
kylemas917 is recommended to review

kylemasc917 commented on calculus.py:
```diff
@@ -370,354 +327,3 @@ def trapezoid_scipy(func, l_lim, u_lim, steps=10000):
     y = func(x)    # evaluate the function on the modified grid
     integral_value = sp.integrate.trapezoid(y, x)   # calculate the integral using numpy
     return integral_value
-
```
Why are these functions being deleted?

kylemasc917 replied:
the undefined pylint errors come from you still using d1,2,3 below but deleting them above. Can be fixed by not deleting them.

kylemasc917 commented on test_calculus.py:
```diff
@@ -7,114 +7,116 @@
+    produces the expected results for certain 
+    integrals.
+    """
+    result1=lib.adapt_c(python_cubic_c, -1,1,10,10)
```
I believe the pytest is failing here because above we've imported the functions as calc so this is missing the calc. before adapt_c

kylemasc917 commented:
Other pylint errors are just some unused imports, some missing docstrings and lines too long, and the deleted functions in calc.py not existing so in testcalc.py when they are called it can't call them.

kylemasc917 reviewed (changes requested):
the code itself you were assigned looks good and runs well, but there are a few changes made to the existing files that are strange and cause a number of errors that can be fixed by not removing certain preexisting functions. Also removing everybody else's functions from the file would cause a number of homework issues and be a real bummer. Once these are addressed, I'll approve, and they can be merged.

abruns123 commented:
Okay, I fixed all pylint/pytest errors. I adjusted my implementation of the wrapper function such that it took in a different parameter type and some of the other functions I made with it became irrelevant. Those were deleted and I adjusted one of the pytests to use the correct parameters. No functions that were deleted were ones I didn't make. I fixed they pytest error with the c_adapt by using calculus. and updating the calculus.dll file via the terminal.

kylemasc917 commented:
Ah ok for some reason the files changed view was showing something crazy but it looks good now. you just have to resolve the merge conflicts

abruns123 commented:
okay, merge conflicts should be resolved now

kylemasc917 reviewed (approved):
Everything looks good. Approved!

kylemasc917 commented:
@zbpetersbuf good to go!
